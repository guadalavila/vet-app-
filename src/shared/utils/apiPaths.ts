export const API_PATHS = {
    SIGN_UP: 'auth/signup',
    LOGIN: 'auth/login',
    GET_ME: 'auth/get-me',
    DASHBOARD: 'dashboard',
    PETS: 'pets',
    ONE_PET: 'pets/one/',
    PETS_BY_VET: 'pets/by-vet/',
    PETS_BY_CLIENT: 'pets/by-client/',
    PET: 'pet',
    CLIENTS: 'clients',
    CLIENTS_BY_VET: '/clients/by-vet/',
    CLIENT: 'clients/',
    SEARCH_CLIENTS: 'clients/search/',
    SEARCH_ONE_CLIENT: 'clients/search/one/',
    VISITS: 'visits',
    VISIT_BY_PET: 'visits/by-pet/',
    LAST_VISIT: 'visits/last/',
    LAST_VISIT_BY_VET: 'visits/by-vet/',
    CONDITIONS: 'conditions',
    PATHOLOGIES: 'pathologies',
    SEARCH: '/search/',
    VACCINE: 'vaccine',
    VACCINE_BY_PET: 'vaccine/by-pet/',
    REPORT_BY_VET: 'report/by-vet/',
    SURGERY: 'surgery',
    SURGERIES_BY_PET: 'surgery/by-pet/',
    VETERINARY: 'vets',
    USERS: 'users',
};
