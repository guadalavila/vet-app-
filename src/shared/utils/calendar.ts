type CONFIG_MONTHS_DAYS_PROPS = {
    monthNames: string[],
    monthNamesShort: string[],
    dayNames: string[],
    dayNamesShort: string[],
    today: string,
};

const monthNames: string[] = [
    'Enero',
    'Febrero',
    'Marzo',
    'Abril',
    'Mayo',
    'Junio',
    'Julio',
    'Agosto',
    'Septiembre',
    'Octubre',
    'Noviembre',
    'Diciembre',
];

const monthNamesShort: string[] = ['Ene', 'Feb', 'Mar', 'Abr', 'May', 'Jun', 'Jul', 'Ago', 'Sep', 'Oct', 'Nov', 'Dic'];

const dayNames: string[] = ['Domingo', 'Lunes', 'Martes', 'Miércoles', 'Jueves', 'Viernes', 'Sábado'];

const dayNamesShort: string[] = ['D', 'L', 'M', 'M', 'J', 'V', 'S'];

const today: string = 'Hoy';

export const LOCALE_CONFIG = 'es';

export const CONFIG_MONTHS_DAYS: CONFIG_MONTHS_DAYS_PROPS = {
    monthNames,
    monthNamesShort,
    dayNames,
    dayNamesShort,
    today,
};
