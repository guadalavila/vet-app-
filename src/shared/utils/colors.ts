import { Theme } from 'react-native-calendars/src/types';
import { typography } from './typography';

const common = {
    // primary: '#FD9340',
    primary: '#917FB3',

    // secondary: '#2A2F4F',
    // tertiary: '#E5BEEC',
    quaternary: '#FDE2F3',
    white: '#FFFFFF',
    black: '#2F2F2F',
    error: '#E74C3C',
    grey: '#EEEEEE',
    greySecondary: '#f5f5f5',
    greyDark: '#A3A3A3',
    whatsapp: '#25d366',
    success: '#4E9F3D',
    teal: '#73B4B0',
};

const light = {
    ...common,
    background: '#FFFFFF',
    text: '#11100F',
    secondary: '#2A2F4F',
    tertiary: '#E5BEEC',
    container: '#f5f4f6',
    separator: '#F1f1f1',
    border: '#2A2F4F',
    input: '#f2f1f6',
    textInput: '#515151',
    bottomSheet: '#F1f1f1',
    card: '#D6D6D6',
    backgroundInput: '#f6f8fe',
};

const dark = {
    ...common,
    background: '#131313',
    text: '#F1F1F1',
    tertiary: '#2A2F4F',
    secondary: '#E5BEEC',
    container: '#222125',
    separator: '#f0f0f0',
    border: '#E5BEEC',
    input: '#4C4C4C',
    textInput: '#DEDEDE',
    bottomSheet: '#71717C',
    card: '#D6D6D6',
    backgroundInput: '#414141',
};

export const colors = { light, dark };

export const THEME_CUSTOM_CALENDAR: Theme = {
    textSectionTitleColor: colors.light.primary,
    textDayHeaderFontSize: typography.size.M,
    calendarBackground: colors.light.grey,
    todayTextColor: colors.light.black,
    arrowHeight: 50,
    textMonthFontSize: typography.size.S,
    textDayHeaderFontWeight: 'bold',
    arrowColor: colors.light.primary,
    dayTextColor: colors.light.black,
    monthTextColor: colors.light.primary,
    selectedDayTextColor: 'white',
    selectedDayBackgroundColor: colors.light.primary,
    textDisabledColor: colors.light.grey,

    // selectedDayBackgroundColor: '#333248',
};
