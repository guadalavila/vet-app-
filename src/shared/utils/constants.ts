import { Categorie } from '../../models/Categorie';

export const CATEGORIES: Categorie[] = [
    { id: 1, name: 'CLIENTES', page: 'ClientsScreen', icon: 'people-outline' },
    { id: 1, name: 'MASCOTAS', page: 'PetsScreen', icon: 'paw-outline' },
];
