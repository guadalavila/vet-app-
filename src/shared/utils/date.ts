export const getPreviousDay = (date = new Date()) => {
    const previous = new Date(date.getTime());
    previous.setDate(date.getDate() - 1);
    return previous;
};

function padTo2Digits(num: number) {
    return num.toString().padStart(2, '0');
}

export const formatDate = (date: Date) => {
    // date.getFullYear()
    return [padTo2Digits(date.getDate()), padTo2Digits(date.getMonth() + 1)].join('/');
};

export const isSelectedDate = (dateA: Date, dateB: Date) => {
    return formatDate(dateA) === formatDate(dateB);
};

export const formatDateCalendar = (date: Date) => {
    return [date.getFullYear(), padTo2Digits(date.getMonth() + 1), padTo2Digits(date.getDate())].join('-');
};

export const getFirstDayOfMonth = () => {
    const date = new Date();
    return new Date(date.getFullYear(), date.getMonth() - 1, 1);
};
