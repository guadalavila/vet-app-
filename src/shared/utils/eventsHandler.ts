class AppEventsHandler<EventType> {
    // eslint-disable-next-line prettier/prettier
    private subscribers: Record<keyof EventType, Set<(data: EventType[any]) => void>> = {} as any;

    listen<K extends keyof EventType>(type: K, cb: (data: EventType[K]) => void): () => void {
        if (!this.subscribers[type]) {
            this.subscribers[type] = new Set([]);
        }
        if (this.subscribers[type].has(cb)) {
            return () => {
                this.subscribers[type].delete(cb);
            };
        }
        this.subscribers[type].add(cb);
        return () => {
            this.subscribers[type].delete(cb);
        };
    }

    publish<K extends keyof EventType>(type: K, data: EventType[K]): void {
        if (!this.subscribers[type]) {
            return;
        }
        this.subscribers[type].forEach((cb) => {
            cb(data);
        });
    }
}
export default AppEventsHandler;
