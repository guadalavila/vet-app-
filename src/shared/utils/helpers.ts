const mapPetType = () => {
    const mapGender = new Map();
    mapGender.set('birds', 'Aves');
    mapGender.set('canine', 'Canino');
    mapGender.set('equine', 'Equino');
    mapGender.set('feline', 'Felino');
    mapGender.set('ferret', 'Huron');
    mapGender.set('insect', 'Insecto');
    mapGender.set('rodent', 'Roedor');
    mapGender.set('unknown', 'Desconocido');
    mapGender.set('', '');
    return mapGender;
};

export const getPetType = (type: string) => {
    return mapPetType().get(type);
};

const mapPetSize = () => {
    const mapSize = new Map();
    mapSize.set('small', 'Chico');
    mapSize.set('medium', 'Mediano');
    mapSize.set('big', 'Grande');
    mapSize.set('other', 'Otro');
    mapSize.set('', '');
    return mapSize;
};

export const getPetSize = (size: string) => {
    return mapPetSize().get(size);
};

const mapPetGender = () => {
    const mapGender = new Map();
    mapGender.set('male', 'Macho');
    mapGender.set('female', 'Hembra');
    mapGender.set('unknow', 'Desconocido');
    mapGender.set('', '');
    return mapGender;
};

export const getPetGender = (gender: string) => {
    return mapPetGender().get(gender);
};
