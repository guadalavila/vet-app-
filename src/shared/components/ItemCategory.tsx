import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { colors } from '../utils/colors';
import { size } from '../utils/size';
import { GlobalStyles } from '../utils/styles';
import { typography } from '../utils/typography';

interface IItemCategoryProps {
    title: string;
    data: number;
    icon: string;
    onPress: () => void;
}

const ItemCategory = ({ title, onPress, icon, data }: IItemCategoryProps) => {
    return (
        <TouchableOpacity style={styles.container} activeOpacity={0.7} onPress={onPress}>
            <View style={[GlobalStyles.selfCenter, styles.containerIcon]}>
                <Icon name={icon} size={38} color={colors.light.primary} />
            </View>
            <Text style={[styles.count]}>{data}</Text>
            <Text style={[styles.category]}>{title}</Text>
        </TouchableOpacity>
    );
};

export default ItemCategory;

const styles = StyleSheet.create({
    container: {
        borderWidth: 2,
        borderColor: colors.light.primary,
        borderRadius: 10,
        margin: size.XS,
        marginHorizontal: size.M,
        paddingTop: size.XXL,
        padding: size.S,
        flex: 1,
        backgroundColor: colors.light.primary,
    },
    category: {
        fontSize: typography.size.M,
        textAlign: 'center',
        fontWeight: '600',
        marginBottom: size.XL,
        color: colors.light.white,
    },
    containerIcon: {
        marginBottom: size.L,
        backgroundColor: colors.light.white,
        borderRadius: 30,
        padding: size.L,
    },
    count: {
        fontSize: typography.size.M,
        textAlign: 'center',
        fontWeight: '500',
        marginVertical: size.XL,
        color: colors.light.secondary,
    },
});
