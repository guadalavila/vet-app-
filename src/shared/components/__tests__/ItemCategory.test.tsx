import React from 'react';
import ItemCategory from '../ItemCategory';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { colors } from '~shared/utils/colors';

describe('ItemCategory', () => {
    const mockFn = jest.fn();

    test('should match snapshot', () => {
        const container = render(<ItemCategory title='Mascotas' data={100} onPress={mockFn} />);
        expect(container).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<ItemCategory title='Mascotas' data={100} onPress={mockFn} />);
        expect(screen.getByTestId('ItemCategory')).toBeTruthy();
    });

    test('should render texts', () => {
        render(<ItemCategory title='Mascotas' data={100} onPress={mockFn} />);
        expect(screen.getByText('Mascotas')).toBeTruthy();
        expect(screen.getByText('100')).toBeTruthy();
    });

    test('should render icon', () => {
        render(<ItemCategory title='Mascotas' data={100} onPress={mockFn} icon='add' />);
        expect(screen.getByTestId('icon')).toBeTruthy();
        expect(screen.getByTestId('icon').props.name).toEqual('add');
        expect(screen.getByTestId('icon').props.size).toEqual(38);
        expect(screen.getByTestId('icon').props.color).toEqual(colors.light.primary);
    });

    test('should call a fn when press component', () => {
        render(<ItemCategory title='Mascotas' data={100} onPress={mockFn} icon='add' />);
        const button = screen.getByTestId('ItemCategory');
        fireEvent.press(button);
        expect(mockFn).toHaveBeenCalled();
    });
});
