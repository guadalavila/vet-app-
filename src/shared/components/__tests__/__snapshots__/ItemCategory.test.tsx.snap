// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCategory should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2A2F4F",
      "borderColor": "#2A2F4F",
      "borderRadius": 10,
      "borderWidth": 2,
      "elevation": 4,
      "flex": 1,
      "height": 250,
      "justifyContent": "space-around",
      "margin": 3,
      "marginBottom": 14,
      "marginHorizontal": 6,
      "opacity": 1,
      "padding": 4,
      "paddingTop": 14,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
  testID="ItemCategory"
>
  <Text
    style={
      Array [
        Object {
          "color": "#F5F5F5",
          "fontSize": 48,
          "fontWeight": "bold",
          "marginVertical": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    100
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "fontWeight": "600",
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Mascotas
  </Text>
</View>
`;
